mypyc/lib-rt/CPy.h

```diff
@@ -936,6 +936,24 @@ static PyObject *CPyIter_Send(PyObject *iter, PyObject *val)
     }
 }
 
+static PyObject *CPy_GetCoro(PyObject *obj)
+{
+    // If the type has an __await__ method, call it,
+    // otherwise, fallback to calling __iter__.
+    PyAsyncMethods* async_struct = obj->ob_type->tp_as_async;
+    if (async_struct != NULL) {
+        if (async_struct->am_await != NULL) {
```

Comment:
If you combine these tests with `&&` we can eliminate the duplication between elses

Reply:
This is done